Automattic/wp-calypso
# Jetpack: Remove the free text entry in the site type step

## Changes proposed in this Pull Request

Resolves #33336 by removing the site type user-input field for Jetpack.

The original question was: _which segment-specific vertical results should we return when a user enters a custom site type?_

Removing the field nullifies that question, and allows us to make changes to the site vertical API results without fretting over whether we're causing cataclysmic destruction to the Jetpack onboarding process. Related changes: Automattic/wp-calypso#33209 D28623-code D28645-code

**Questions for @Automattic/luna** 

Have I missed anything?

Any issues with deleting `user_input_site_type` from the `calypso_signup_actions_submit_site_type` tracks event?

🙇 

## Testing instructions

Check that there is no regression in WordPress.com

http://calypso.localhost:3000/start/site-type

<img width="515" alt="Screen Shot 2019-05-27 at 12 25 39 pm" src="https://user-images.githubusercontent.com/6458278/58392156-f7e7b100-807b-11e9-9c95-46dbe47c3e1c.png">

Connect a Jetpack site and proceed through the onboarding flow:

<img width="612" alt="Screen Shot 2019-05-27 at 12 34 20 pm" src="https://user-images.githubusercontent.com/6458278/58392157-f7e7b100-807b-11e9-9838-6f7388ebb2a5.png">
